libspeedhack.cpp:222:16: error: conflicting declaration of C function ‘int gettimeofday(timeval*, timezone*)’

```
mkdir -p lib64
g++ -std=gnu++14 -Wall -m64 -fPIC -shared -Wl,-init=init_libspeedhack libspeedhack.cpp -ldl -o lib64/libspeedhack.so
libspeedhack.cpp:222:16: error: conflicting declaration of C function ‘int gettimeofday(timeval*, timezone*)’
  222 | extern "C" int gettimeofday(timeval *tv, struct timezone *tz)
      |                ^~~~~~~~~~~~
In file included from libspeedhack.cpp:3:
/usr/include/sys/time.h:66:12: note: previous declaration ‘int gettimeofday(timeval*, void*)’
   66 | extern int gettimeofday (struct timeval *__restrict __tv,
      |            ^~~~~~~~~~~~
libspeedhack.cpp: In function ‘time_t time(time_t*)’:
libspeedhack.cpp:290:24: error: call of overloaded ‘gettimeofday(timeval*, NULL)’ is ambiguous
  290 |  gettimeofday(&tv, NULL);
      |                        ^
In file included from libspeedhack.cpp:3:
/usr/include/sys/time.h:66:12: note: candidate: ‘int gettimeofday(timeval*, void*)’
   66 | extern int gettimeofday (struct timeval *__restrict __tv,
      |            ^~~~~~~~~~~~
libspeedhack.cpp:222:16: note: candidate: ‘int gettimeofday(timeval*, timezone*)’
  222 | extern "C" int gettimeofday(timeval *tv, struct timezone *tz)
      |                ^~~~~~~~~~~~
libspeedhack.cpp: At global scope:
libspeedhack.cpp:112:17: warning: ‘timespec operator*(double, const timespec&)’ defined but not used [-Wunused-function]
  112 | static timespec operator * (double x, const timespec &tv)
      |                 ^~~~~~~~
libspeedhack.cpp:54:16: warning: ‘timeval operator*(double, const timeval&)’ defined but not used [-Wunused-function]
   54 | static timeval operator * (double x, const timeval &tv)
      |                ^~~~~~~~
make: *** [Makefile:13: lib64/libspeedhack.so] Error 1
```

--- a/libspeedhack.cpp
+++ b/libspeedhack.cpp
@@ -137,7 +137,7 @@ static timespec
 	_monotoniczero, _monotoniccoarsezero, _monotonicrawzero,
 	_boottimezero, _processzero, _threadzero;
 
-static int (*gettimeofday_orig)(timeval *tv, struct timezone *tz);
+static int (*gettimeofday_orig)(timeval *tv, void *tz);
 static int (*clock_gettime_orig)(clockid_t clk_id, timespec *tp);
 
 static double speedup = 1;
@@ -219,7 +219,7 @@ extern "C" void init_libspeedhack()
 	fix_timescale();
 }
 
-extern "C" int gettimeofday(timeval *tv, struct timezone *tz)
+extern "C" int gettimeofday(timeval *tv, void *tz)
 {
 	if (!inited)
 		init_libspeedhack();

